src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerUdpBatcher.cs

```diff
@@ -87,7 +95,50 @@ public JaegerUdpBatcher(JaegerExporterOptions options, TTransport clientTranspor
 
         internal Dictionary<string, Batch> CurrentBatches { get; } = new Dictionary<string, Batch>();
 
-        public async ValueTask<int> AppendAsync(JaegerSpan jaegerSpan, CancellationToken cancellationToken)
+        public async ValueTask<int> AppendBatchAsync(IEnumerable<Activity> activityBatch, CancellationToken cancellationToken)

```

Comment:
This is an optimization in JaegerUdpBatcher. Before it would acquire the lock for each Span/Activity in the batch. I moved it up a level higher. Should improve perf.